### ISSUE TITLE
Handle more resource kind health status

### ISSUE THREAD

**Issue by undera** [komodorio/helm-dashboard#418]
In backend call like `/api/helm/releases/:ns/:name/resources?health=true`, we need to support more resource kinds:
 - ExternalSecret
 - Job

And generally we need to handle unknown object kinds to at least show "Exists" status.

**Comment by iamvikaskumar** [comment]
Hi,
I would be happy to work on this.

**Comment by undera** [comment]
Feel free to bring PR @iamvikaskumar

**Comment by iamvikaskumar** [comment]
Thanks @undera . Could you please provide some more details about the issue. 
My understanding is as below, please let me know if I got it rite :
The issue is, that for the above mentioned resource kinds, we are not calculating hdHealth ?
Below is an example of a CRD where status is unknown.
<img width="410" alt="image" src="https://github.com/komodorio/helm-dashboard/assets/35912164/64027a55-d6b6-498b-8113-5c7eb3745e38">

**Comment by iamvikaskumar** [comment]
Please let me know if I got it wrong ?

**Comment by undera** [comment]
Yes, your understanding is correct. We want to have more meaningful `hdHealth` for each resource, at least checking that it exists.

**Comment by iamvikaskumar** [comment]
Thanks for your reply @undera . I have one more doubt before I raise the PR... when a resource have more than one condition e.g. in case of CRD (refer the image below), do we mark hdHealth as healthy only if all the conditions has status as "True" ?

<img width="500" alt="image" src="https://github.com/komodorio/helm-dashboard/assets/35912164/f4225078-b6ab-40cf-a9b5-ac3cda92eebd">

**Comment by undera** [comment]
There is no strict rule to set `hdHealth` based on conditions. Conditions in k8s are too broad in their meaning. So to calculate our `hdHealth` helper, we have roughly following rules:
1. If resource does not exist in cluster - it's definitely "NotFound" and unhealthy.
2. If we have a dedicated custom code for that resource kind - we try to calculate "progressing" yellow status, or more meaningful "Unhealthy/Healthy" status.
3. Worst case, we fall back to the "Exists/NotFound" logic.

So for your example of CRD above, the custom logic _may_ be to have both conditions True. But that is only for CustomResourceDefinition, and not general rule.

**Comment by alessandrodetta** [comment]
@undera I tried deleting a resource with `kubectl delete namespace|deployment <name>`, to see what kind of status could I get for it but I didn't expect to get the unknown status, but "NotFound" as you mentioned in your last answer. 

![image](https://github.com/komodorio/helm-dashboard/assets/62752370/881d3057-9d3e-4094-895e-b56b55d87a2b)

Should we change the code to broadly follow something like the idea presented below? Or there is already some lines of code that actually checks for "NotFound", which I missed?

```
...
// custom logic to provide most meaningful status for the resource
	if err != nil {
		if errors.IsNotFound(err) {
			c.Status = Unhealthy
			c.Reason = "NotFound"
			c.Message = err.Error()
		} else {
			c.Status = Unknown// Or Unhealthy anyway? 
			c.Reason = "ErrorGettingStatus"
			c.Message = err.Error()
		}
        ...
```

**Comment by undera** [comment]
@alessandrodetta Your proposition is correct. If we know the resource does not exist - we should set status to that, it's better than generic failure

**Comment by visionaryfire** [comment]
@undera Is this issue still valid one for the contribution?

**Comment by undera** [comment]
@visionaryfire I believe so
